lib/elixir/lib/calendar/datetime.ex

```diff
@@ -998,6 +1005,62 @@ defmodule DateTime do
     naive_diff + System.convert_time_unit(offset_diff, :second, unit)
   end
 
+  @doc """
+  Adds a specified amount of time to a `DateTime`.
+
+  Accepts an `integer` in any `unit` available from `t:System.time_unit/0`.
+  Negative values will move backwards in time.
+
+  Takes changes such as summer time/DST into account. This means that adding time
+  can cause the wall time to "go backwards" during "fall back" during autumn.
+  Adding just a few seconds to a datetime just before "spring forward" can cause wall
+  time to increase by more than an hour.
+
+  Fractional second precision stays the same in a similar way to `NaiveDateTime.add/2`.
+
+  ### Examples
+
+      iex> dt = DateTime.from_naive!(~N[2018-11-15 10:00:00], "Europe/Copenhagen", FakeTimeZoneDatabase)
+      iex> dt |> DateTime.add(3600, :second, FakeTimeZoneDatabase)
+      #DateTime<2018-11-15 11:00:00+01:00 CET Europe/Copenhagen>
+
+      iex> dt = DateTime.from_naive!(~N[2018-11-15 10:00:00], "Etc/UTC")
+      iex> dt |> DateTime.add(3600, :second)
+      #DateTime<2018-11-15 11:00:00Z>
+
+  When adding 3 seconds just before "spring forward" we go from 1:59:59 to 3:00:02
+
+      iex> dt = DateTime.from_naive!(~N[2019-03-31 01:59:59.123], "Europe/Copenhagen", FakeTimeZoneDatabase)
+      iex> dt |> DateTime.add(3, :second, FakeTimeZoneDatabase)
+      #DateTime<2019-03-31 03:00:02.123+02:00 CEST Europe/Copenhagen>
+
+  """
+  @doc since: "1.8.0"
+  @spec add(t, integer, System.time_unit(), Calendar.time_zone_database()) :: t()
+  def add(
+        %DateTime{
```

Comment:
Minor nitpicks: 

1. move variable assignment to the first line
2. rename `integer_to_add` to `integer` (same name variable as `NaiveDateTime.add/3`)

Reply:
Good idea. I have renamed the variable in both functions to `amount_to_add` for consistency and a better description of what the variable is for.